agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/TelemetryTruncation.java

```diff
@@ -21,28 +21,60 @@
 
 package com.microsoft.applicationinsights.agent.internal.common;
 
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class TelemetryTruncation {
 
-  private static final WarningLogger warningLogger =
-      new WarningLogger(TelemetryTruncation.class, "Telemetry was truncated");
+  private static final Logger logger = LoggerFactory.getLogger(TelemetryTruncation.class);
+
+  private static final Set<String> alreadyLoggedAttributeNames = ConcurrentHashMap.newKeySet();
+  private static final Set<String> alreadyLoggedPropertyKeys = ConcurrentHashMap.newKeySet();
 
   public static String truncateTelemetry(String value, int maxLength, String attributeName) {
     if (value == null || value.length() <= maxLength) {
       return value;
     }
-    warningLogger.recordWarning("truncated " + attributeName + ": " + value);
+    if (!alreadyLoggedAttributeNames.add(attributeName)) {
+      // this can be expected, so don't want to flood the logs with a lot of these
+      // (and don't want to log the full value, e.g. sql text > 8192 characters)
+      logger.warn(
```

Comment:
should we continue to use the AggregatingLogger? it will log for the first 5 mins..  
i think aggregation is still helpful for other customers..  when truncation happens occasionally.

Reply:
i'm not sure i follow "it will log for the first 5 mins"?

Reply:
it's using a scheduler to run every 5 mins.  it's not possible to log once using aggregated logger unless to stop the scheduler, but we don't have any condition to stop it..  i see why switch from aggregated logger to regular logger.